components/StorageFileListItem.vue

```diff
@@ -1,13 +1,265 @@
 <script setup lang="ts">
 import { FileObject } from "@supabase/storage-js";
 
+const XlsxFile = resolveComponent("XlsxFile");
+const SomeFile = resolveComponent("SomeFile");
+const DocxFile = resolveComponent("DocxFile");
+
+const client = useSupabaseClient();
+
+const layoutType = useLayoutType();
+
 const props = defineProps<{
   file: FileObject;
 }>();
+
+const fileName = computed(() => {
+  return props.file.name;
+});
+
+const fileSize = computed(() => {
+  const size = props.file.metadata.size;
+  const sizeLength = size.toString().length;
+  if (sizeLength <= 5) {
+    return `size: ${(size / 1000).toFixed(1)}KB`;
+  } else {
+    return `size: ${(size / 1000000).toFixed(1)}MB`;
+  }
+});
+
+const sortFileType = computed(() => {
+  const type = props.file.metadata.mimetype.split("/")[0];
+  return `type: ${type}`;
+});
+
+const previewFileType = computed(() => {
+  const type = props.file.metadata.mimetype;
+  const explicitTypes = type.split("/")[0];
+
+  const splitName = props.file.name.split(".");
+  const fileExtension = splitName[splitName.length - 1];
```

Comment:
hint: you can use `array.at(-1)` to get last element in the array 👌 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at

Reply:
Wow this is great, I was not familiar with this method :), thank you for showing me :)